app/src/main/java/org/simple/clinic/appupdate/criticalupdatedialog/CriticalAppUpdateUiRenderer.kt

```diff
@@ -12,5 +12,8 @@ class CriticalAppUpdateUiRenderer(
     } else {
       ui.hideHelp()
     }
+
+    if (model.isCriticalUpdateNudgePriority && model.hasAppStaleness)
+      ui.renderCriticalAppUpdateReason(model.appStaleness!!)
```

Comment:
@msasikanth what are your thoughts about having two different methods for rendering the reason in the dialog? I thought of doing it this way because in this method we will also have to manipulate the string a bit according to the app staleness value and in the other its going to be one particular string. Although, we can have a single method for this. Would also like some opinion on the name. Bit unsure about it.

Reply:
I am inclining towards having a single ui renderer method which handles the rest of the logic.

Reply:
We can do the check in the UI renderer? That way we can have a UI renderer test for this case as well. WDYT?

Reply:
That's how I have implemented it, currently. The string manipulation that we are going to based on `appStaleness` for example writing 3 months or 23 days is a very ui based logic which I think we don't need to add in the renderer.

Reply:
Instead of passing the `appStaleness`, could we construct the `LocalDate` or `Duration`, since that is what we needed for rendering? If it's a critical security priority update, we can just render the other message. That way we don't have to construct the date or duration in the UI. Thoughts?

Reply:
👍🏻